### DISCUSSION TITLE
How to fire event using @click when clicking on text in a Text element?

### DISCUSSION THREAD

**Discussion by bradparks** [Textualize/textual#323] [Q&A]
I'm trying to get it so I can add links in text rendered by Textual, as [discussed in this thread in rich discussions](https://github.com/Textualize/rich/discussions/2021#discussioncomment-2288880) which led me to textual (great!).

but... I don't seem to be able to get it to work. So I built this little sample to try it out. 

Anyone know what I'm doing wrong here?

In this sample, clicking on the word "World" should change the background color to red.

NOTE: I also bound the "b" key to do pretty much the same thing, so I could see it work
It should change the background color, and the subtitle of the app.
Thanks for any pointers!

``` python
import os
import sys
from rich.console import RenderableType
from rich.panel import Panel
from rich.text import Text
from textual.app import App
from textual.widgets import Header, Footer, ScrollView
from textual.widgets import Placeholder

class MyApp(App):

    async def on_load(self) -> None:
        await self.bind("b", "color('blue')")

    async def on_mount(self) -> None:
        await self.view.dock(Header(), size=5, edge="top")
        await self.view.dock(Footer(), edge="bottom")
        await self.view.dock(ScrollView(Panel("Hello [@click=hello]World[/], [red]more[/] text")), edge="top")

    async def action_color(self, color:str) -> None:
        self.app.sub_title = "KEYBOARD"
        self.background = f"on {color}"

    async def action_hello(self) -> None:
        self.app.sub_title = "CLICKED"
        self.background = "on red"

MyApp.run(title="Test click", log="textual.log")


```

**Comment by bradparks** [comment]
any ideas here? I'm sure I'm missing something simple... much appreciated!

**Comment by bradparks** [comment]
NOTE: I posted this to stackoverflow as well, with a bounty of 200, to see if someone could see how this is possible, fyi:

https://stackoverflow.com/questions/71433131/textual-python-how-to-add-click-event-in-simple-text-object

**Comment by heeplr** [comment]
I found a solution that works here: https://github.com/jspv/spotbot/blob/main/spotbot/Widgets/footer.py#L70

I do this:
```python
text = Text.from_markup("https://textualize.io")
text.apply_meta(
    meta={
        "@mouse.down": f"app.press('b')",
        "link_id": my_link_id
    }
)
```

and in the parent widget::
```python
    async def on_mouse_down(self, event: events.MouseDown) -> None:
        self.link_id = event.style.meta.get("link_id")
```

This emulates a "b" press when the text is clicked. ```meta``` can hold any kind of key/value to differ your various links.
